registry.ts

```diff
@@ -1,9 +1,9 @@
-// FIXME don't use any[]
-export function lookup(url: string, registries: any[]):
+type RegistryCtor = new (url: string) => RegistryUrl;
```

Comment:
What does the `new` do here? I'm unfamiliar with this pattern. Could you explain it?

Reply:
It means `RegistryCtor` is a constructor function type, rather than just an ordinary function type.

Without this TypeScript complains on line 6 when we do
```typescript
const u = new R(url);
```
that the type of `R` "lacks a construct signature"